Mathlib/Algebra/MvPolynomial/Equiv.lean

```diff
@@ -318,40 +318,257 @@ lemma optionEquivRight_X_none : optionEquivRight R S₁ (X none) = C Polynomial.
+    finSuccEquivNth R p (X i.succ) = Polynomial.C (X i) := by
+  simp [finSuccEquivNth_apply, Fin.insertNth_apply_above h _ _]
+
+theorem finSuccEquivNth_X_below {i : Fin n} (h : i.castSucc < p) :
+    finSuccEquivNth R p (X i.castSucc) = Polynomial.C (X i) := by
+  simp [finSuccEquivNth_apply, Fin.insertNth_apply_below h _ _]
+
+/-- The coefficient of `m` in the `i`-th coefficient of `finSuccEquivNth R p f` equals the
+    coefficient of `m.insertNth p i` in `f`. -/
+theorem finSuccEquivNth_coeff_coeff (m : Fin n →₀ ℕ) (f : MvPolynomial (Fin (n + 1)) R) (i : ℕ) :
+    coeff m (Polynomial.coeff (finSuccEquivNth R p f) i) = coeff (m.insertNth p i) f := by
+  induction' f using MvPolynomial.induction_on' with u a p q hp hq generalizing i m
+  · simp only [finSuccEquivNth_apply, coe_eval₂Hom, eval₂_monomial, RingHom.coe_comp, comp_apply,
+      prod_pow, Fin.prod_univ_succAbove _ p, Fin.insertNth_apply_same,
+      Fin.insertNth_apply_succAbove, Polynomial.coeff_C_mul, coeff_C_mul, coeff_monomial,
+      ← map_prod, ← RingHom.map_pow]
+    rw [← mul_boole, mul_comm (Polynomial.X ^ u p), Polynomial.coeff_C_mul_X_pow]; congr 1
+    obtain rfl | hjmi := eq_or_ne u (m.insertNth p i)
+    · simpa only [insertNth_apply_same, if_pos rfl, insertNth_apply_succAbove, monomial_eq, C_1,
+        one_mul, prod_pow] using coeff_monomial m m (1 : R)
+    · simp only [hjmi, if_false]
+      obtain hij | rfl := ne_or_eq i (u p)
+      · simp only [hij, if_false, coeff_zero]
+      simp only [eq_self_iff_true, if_true]
+      have hmj : m ≠ u.removeNth p := by
+        rintro rfl
+        rw [insertNth_self_removeNth] at hjmi
+        contradiction
+      simpa only [monomial_eq, C_1, one_mul, prod_pow, Finsupp.removeNth_apply, if_neg hmj.symm]
+        using coeff_monomial m (u.removeNth p) (1 : R)
+  · simp only [map_add, Polynomial.coeff_add, coeff_add, hp, hq]
+
+/-- The evaluation of `f` at `Fin.insertNth p y s` equals the evaluation at `y` of the polynomial
+obtained by partially evaluating `finSuccEquivNth R p f` at `s`.
+-/
+theorem eval_eq_eval_mv_eval_finSuccEquivNth (s : Fin n → R) (y : R)
+    (f : MvPolynomial (Fin (n + 1)) R) :
+      eval (Fin.insertNth p y s : Fin (n + 1) → R) f =
+        Polynomial.eval y (Polynomial.map (eval s) (finSuccEquivNth R p f)) := by
+  show
+    aeval (Fin.insertNth p y s : Fin (n + 1) → R) f = (Polynomial.aeval y).comp
+      ((Polynomial.mapAlgHom (aeval s)).comp (finSuccEquivNth R p).toAlgHom) f
+  congr 2
+  apply MvPolynomial.algHom_ext
+  rw [Fin.forall_iff_succAbove p]
+  simp only [aeval_X, Fin.insertNth_apply_same, Polynomial.mapAlgHom, AlgHom.toRingHom_eq_coe,
+    coe_aeval_eq_eval, AlgEquiv.toAlgHom_eq_coe, AlgHom.coe_comp, Polynomial.coe_aeval_eq_eval,
+    AlgHom.coe_mk, coe_mapRingHom, AlgHom.coe_coe, comp_apply, finSuccEquivNth_apply, eval₂Hom_X',
+    Polynomial.map_X, Polynomial.eval_X, Fin.insertNth_apply_succAbove, Polynomial.map_C, eval_X,
+    Polynomial.eval_C, implies_true, and_self]
```

Comment:
Since this is the last statement in a proof block, could it be changed to `simp [Fin.forall_iff_succAbove p]`? (untested)